From 25ac19d2aeb82bf2148ca7e1f716746a457b7f11 Mon Sep 17 00:00:00 2001
From: David Li <davidli.li@linux.alibaba.com>
Date: Sun, 1 Sep 2024 00:25:14 +0800
Subject: [PATCH 193/400] audio: th1520: add soundcard dts node of th1520-a-val
 board

Add soundcard dts nodes of th1520-a-val board, that contains
4 types of soundcards:
    soundcard with i2s-2ch
	soundcard with i2s-8ch
	soundcard with tdm
	soudncard with spdif
you could enable each soundcard while selected related dts file

Signed-off-by: David Li <davidli.li@linux.alibaba.com>
Signed-off-by: Han Gao <gaohan@iscas.ac.cn>
---
 arch/riscv/boot/dts/thead/Makefile            |   1 +
 .../dts/thead/th1520-a-val-audio-i2s-8ch.dts  |  67 ++++
 .../dts/thead/th1520-a-val-audio-spdif.dts    |  18 +
 .../boot/dts/thead/th1520-a-val-audio-tdm.dts |  62 ++++
 .../boot/dts/thead/th1520-a-val-audio.dts     |  40 +++
 .../boot/dts/thead/th1520-a-val-audio.dtsi    | 334 ++++++++++++++++++
 arch/riscv/boot/dts/thead/th1520-a-val.dts    |  39 ++
 arch/riscv/boot/dts/thead/th1520-a-val.dtsi   | 123 +++++--
 sound/soc/xuantie/th1520-i2s-8ch.c            |   1 +
 9 files changed, 650 insertions(+), 35 deletions(-)
 create mode 100644 arch/riscv/boot/dts/thead/th1520-a-val-audio-i2s-8ch.dts
 create mode 100644 arch/riscv/boot/dts/thead/th1520-a-val-audio-spdif.dts
 create mode 100644 arch/riscv/boot/dts/thead/th1520-a-val-audio-tdm.dts
 create mode 100644 arch/riscv/boot/dts/thead/th1520-a-val-audio.dts
 create mode 100644 arch/riscv/boot/dts/thead/th1520-a-val-audio.dtsi

diff --git a/arch/riscv/boot/dts/thead/Makefile b/arch/riscv/boot/dts/thead/Makefile
index f83afc00a41b..570940141863 100644
--- a/arch/riscv/boot/dts/thead/Makefile
+++ b/arch/riscv/boot/dts/thead/Makefile
@@ -2,3 +2,4 @@
 dtb-$(CONFIG_ARCH_XUANTIE) += th1520-lichee-pi-4a.dtb th1520-beaglev-ahead.dtb th1520-lichee-pi-4a-16g.dtb
 dtb-$(CONFIG_ARCH_XUANTIE) += th1520-lpi4a-product.dtb th1520-lpi4a-product-sec.dtb
 dtb-$(CONFIG_ARCH_XUANTIE) += th1520-a-val.dtb th1520-a-val-sec.dtb
+dtb-$(CONFIG_ARCH_XUANTIE) += th1520-a-val-audio.dtb th1520-a-val-audio-i2s-8ch.dtb th1520-a-val-audio-tdm.dtb th1520-a-val-audio-spdif.dtb
diff --git a/arch/riscv/boot/dts/thead/th1520-a-val-audio-i2s-8ch.dts b/arch/riscv/boot/dts/thead/th1520-a-val-audio-i2s-8ch.dts
new file mode 100644
index 000000000000..ab32a8118b33
--- /dev/null
+++ b/arch/riscv/boot/dts/thead/th1520-a-val-audio-i2s-8ch.dts
@@ -0,0 +1,67 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/*
+ * Copyright (C) 2021 Alibaba Group Holding Limited.
+ */
+
+#include "th1520-a-val.dtsi"
+
+/ {
+	model = "XuanTie TH1520 FM Audio VAL board";
+	compatible = "xuantie,th1520-val-audio-i2s-8ch", "xuantie,th1520";
+};
+
+&th1520_sound_i2s_8ch {
+	status = "okay";
+};
+
+&i2s0 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2s0_pa_pins>;
+};
+
+&i2s_8ch_sd0 {
+	status = "okay";
+    pinctrl-names = "default";
+	pinctrl-0 = <&i2s_8ch_bus_pa_pins>,
+				<&i2s_8ch_sd0_pa_pins>;
+};
+
+&i2s_8ch_sd1 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2s_8ch_bus_pa_pins>,
+				<&i2s_8ch_sd1_pa_pins>;
+};
+
+&i2s_8ch_sd2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2s_8ch_bus_pa_pins>,
+				<&i2s_8ch_sd2_pa_pins>;
+};
+
+&i2s_8ch_sd3 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2s_8ch_bus_pa_pins>,
+				<&i2s_8ch_sd3_pa_pins>;
+};
+
+&es8156_audio_codec {
+	status = "okay";
+};
+
+&es7210_audio_codec_adc0 {
+	status = "okay";
+	channels-max = <8>;
+};
+
+&es7210_audio_codec_adc1 {
+	status = "okay";
+	channels-max = <8>;
+};
+
+&audio_aw87519_pa {
+	status = "okay";
+};
diff --git a/arch/riscv/boot/dts/thead/th1520-a-val-audio-spdif.dts b/arch/riscv/boot/dts/thead/th1520-a-val-audio-spdif.dts
new file mode 100644
index 000000000000..55d79741c079
--- /dev/null
+++ b/arch/riscv/boot/dts/thead/th1520-a-val-audio-spdif.dts
@@ -0,0 +1,18 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/*
+ * Copyright (C) 2021 Alibaba Group Holding Limited.
+ */
+
+#include "th1520-a-val.dtsi"
+
+&spdif0 {
+	status = "okay";
+};
+
+&spdif1 {
+	status = "okay";
+};
+
+&th1520_sound_spdif {
+	status = "okay";
+};
diff --git a/arch/riscv/boot/dts/thead/th1520-a-val-audio-tdm.dts b/arch/riscv/boot/dts/thead/th1520-a-val-audio-tdm.dts
new file mode 100644
index 000000000000..e145cc4dfdca
--- /dev/null
+++ b/arch/riscv/boot/dts/thead/th1520-a-val-audio-tdm.dts
@@ -0,0 +1,62 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/*
+ * Copyright (C) 2021 Alibaba Group Holding Limited.
+ */
+
+#include "th1520-a-val.dtsi"
+
+&tdm_slot1 {
+	status = "okay";
+};
+
+&tdm_slot2 {
+	status = "okay";
+};
+
+&tdm_slot3 {
+	status = "okay";
+};
+
+&tdm_slot4 {
+	status = "okay";
+};
+
+&tdm_slot5 {
+	status = "okay";
+};
+
+&tdm_slot6 {
+	status = "okay";
+};
+
+&tdm_slot7 {
+	status = "okay";
+};
+
+&tdm_slot8 {
+	status = "okay";
+};
+
+&es8156_audio_codec {
+	status = "okay";
+};
+
+&es7210_adc2 {
+	status = "okay";
+};
+
+&es7210_adc3 {
+	status = "okay";
+};
+
+&th1520_sound_tdm {
+	status = "okay";
+};
+
+&i2s0 {
+	status = "okay";
+};
+
+&audio_aw87519_pa {
+	status = "okay";
+};
diff --git a/arch/riscv/boot/dts/thead/th1520-a-val-audio.dts b/arch/riscv/boot/dts/thead/th1520-a-val-audio.dts
new file mode 100644
index 000000000000..8c73b999bf28
--- /dev/null
+++ b/arch/riscv/boot/dts/thead/th1520-a-val-audio.dts
@@ -0,0 +1,40 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/*
+ * Copyright (C) 2021 Alibaba Group Holding Limited.
+ */
+
+#include "th1520-a-val.dtsi"
+
+/ {
+	model = "XuanTie TH1520 FM Audio VAL board";
+	compatible = "xuantie,th1520-val-audio", "xuantie,th1520";
+};
+
+&th1520_sound {
+	status = "okay";
+};
+
+&i2s0 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2s0_pa_pins>;
+};
+
+&i2s_8ch_sd2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2s_8ch_bus_pa_pins>,
+				<&i2s_8ch_sd2_pa_pins>;
+};
+
+&es8156_audio_codec {
+	status = "okay";
+};
+
+&es7210_audio_codec_adc0 {
+	status = "okay";
+};
+
+&audio_aw87519_pa {
+	status = "okay";
+};
diff --git a/arch/riscv/boot/dts/thead/th1520-a-val-audio.dtsi b/arch/riscv/boot/dts/thead/th1520-a-val-audio.dtsi
new file mode 100644
index 000000000000..5149b142416f
--- /dev/null
+++ b/arch/riscv/boot/dts/thead/th1520-a-val-audio.dtsi
@@ -0,0 +1,334 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/*
+ * Copyright (C) 2021-2022 Alibaba Group Holding Limited.
+ */
+
+/ {
+	th1520_sound: soundcard@1 {
+		compatible = "simple-audio-card";
+		simple-audio-card,name = "TH1520-Sound-Card";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		status = "disabled";
+
+		simple-audio-card,widgets = "Speaker", "Speaker";
+		simple-audio-card,routing =
+			"Speaker", "AW87519 VO",
+			"AW87519 IN", "ES8156 ROUT";
+		simple-audio-card,aux-devs = <&audio_aw87519_pa>;
+		simple-audio-card,dai-link@0 {          /* I2S - AUDIO SYS CODEC 8156*/
+				reg = <0>;
+				format = "i2s";
+				cpu {
+						sound-dai = <&i2s0 0>;
+				};
+				codec {
+						sound-dai = <&es8156_audio_codec>;
+				};
+		};
+
+		simple-audio-card,dai-link@1 {          /* I2S - AUDIO SYS CODEC 7210*/
+				reg = <1>;
+				format = "i2s";
+				cpu {
+						sound-dai = <&i2s_8ch_sd2 2>;
+				};
+				codec {
+						sound-dai = <&es7210_audio_codec_adc0>;
+				};
+		};
+	};
+
+	th1520_sound_spdif: soundcard@2 {
+		compatible = "simple-audio-card";
+		simple-audio-card,name = "TH1520-Sound-Card";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		status = "disabled";
+
+		simple-audio-card,dai-link@0 {          /* SPDIF0 */
+				reg = <1>;
+				format = "i2s";
+				cpu {
+						sound-dai = <&spdif0>;
+				};
+				codec {
+						sound-dai = <&hdmi_codec>;
+				};
+		};
+		simple-audio-card,dai-link@1 {          /* SPDIF1 */
+				reg = <1>;
+				format = "i2s";
+				cpu {
+						sound-dai = <&spdif1>;
+				};
+				codec {
+						sound-dai = <&hdmi_codec>;
+				};
+		};
+	};
+
+	th1520_sound_tdm: soundcard@3 {
+		compatible = "simple-audio-card";
+		simple-audio-card,name = "TH1520-Sound-Card";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		status = "disabled";
+
+		simple-audio-card,widgets = "Speaker", "Speaker";
+		simple-audio-card,routing =
+			"Speaker", "AW87519 VO",
+			"AW87519 IN", "ES8156 ROUT";
+		simple-audio-card,aux-devs = <&audio_aw87519_pa>;
+		simple-audio-card,dai-link@0 {          /* I2S - AUDIO SYS CODEC 8156*/
+				reg = <0>;
+				format = "i2s";
+				cpu {
+						sound-dai = <&i2s0 0>;
+				};
+				codec {
+						sound-dai = <&es8156_audio_codec>;
+				};
+		};
+
+		simple-audio-card,dai-link@1 {          /* TDM - AUDIO SYS CODEC 7210*/
+				reg = <1>;
+				format = "dsp_b";
+				cpu {
+						sound-dai = <&tdm_slot1>;
+				};
+				codec {
+						sound-dai = <&es7210_adc2>;
+				};
+		};
+
+		simple-audio-card,dai-link@2 {
+				reg = <1>;
+				format = "dsp_b";
+				cpu {
+						sound-dai = <&tdm_slot2>;
+				};
+				codec {
+						sound-dai = <&es7210_adc2>;
+				};
+		};
+
+		simple-audio-card,dai-link@3 {
+				reg = <1>;
+				format = "dsp_b";
+				cpu {
+						sound-dai = <&tdm_slot3>;
+				};
+				codec {
+						sound-dai = <&es7210_adc2>;
+				};
+		};
+
+		simple-audio-card,dai-link@4 {
+				reg = <1>;
+				format = "dsp_b";
+				cpu {
+						sound-dai = <&tdm_slot4>;
+				};
+				codec {
+						sound-dai = <&es7210_adc2>;
+				};
+		};
+
+		simple-audio-card,dai-link@5 {
+				reg = <1>;
+				format = "dsp_b";
+				cpu {
+						sound-dai = <&tdm_slot5>;
+				};
+				codec {
+						sound-dai = <&es7210_adc2>;
+				};
+		};
+
+		simple-audio-card,dai-link@6 {
+				reg = <1>;
+				format = "dsp_b";
+				cpu {
+						sound-dai = <&tdm_slot6>;
+				};
+				codec {
+						sound-dai = <&es7210_adc2>;
+				};
+		};
+
+		simple-audio-card,dai-link@7 {
+				reg = <1>;
+				format = "dsp_b";
+				cpu {
+						sound-dai = <&tdm_slot7>;
+				};
+				codec {
+						sound-dai = <&es7210_adc2>;
+				};
+		};
+
+		simple-audio-card,dai-link@8 {
+				reg = <1>;
+				format = "dsp_b";
+				cpu {
+						sound-dai = <&tdm_slot8>;
+				};
+				codec {
+						sound-dai = <&es7210_adc2>;
+				};
+		};
+	};
+
+	th1520_sound_i2s_8ch: soundcard@4 {
+		compatible = "simple-audio-card";
+		simple-audio-card,name = "TH1520-Sound-Card";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		status = "disabled";
+
+		simple-audio-card,widgets = "Speaker", "Speaker";
+		simple-audio-card,routing =
+			"Speaker", "AW87519 VO",
+			"AW87519 IN", "ES8156 ROUT";
+		simple-audio-card,aux-devs = <&audio_aw87519_pa>;
+		simple-audio-card,dai-link@0 {          /* I2S - AUDIO SYS CODEC 8156*/
+				reg = <0>;
+				format = "i2s";
+				cpu {
+						sound-dai = <&i2s0 0>;
+				};
+				codec {
+						sound-dai = <&es8156_audio_codec>;
+				};
+		};
+
+		simple-audio-card,dai-link@1 {          /* I2S - AUDIO SYS CODEC 7210*/
+				reg = <1>;
+				format = "i2s";
+				cpu {
+						sound-dai = <&i2s_8ch_sd2>;
+				};
+				codec {
+						mclk-fs = <512>;
+						sound-dai = <&es7210_audio_codec_adc0>;
+				};
+		};
+
+		simple-audio-card,dai-link@2 {          /* I2S - AUDIO SYS CODEC 7210*/
+				reg = <2>;
+				format = "i2s";
+				cpu {
+						sound-dai = <&i2s_8ch_sd3>;
+				};
+				codec {
+						mclk-fs = <512>;
+						sound-dai = <&es7210_audio_codec_adc0>;
+				};
+		};
+
+		simple-audio-card,dai-link@3 {          /* I2S - AUDIO SYS CODEC 7210_1*/
+				reg = <3>;
+				format = "i2s";
+				cpu {
+						sound-dai = <&i2s_8ch_sd0>;
+				};
+				codec {
+						mclk-fs = <512>;
+						sound-dai = <&es7210_audio_codec_adc1>;
+				};
+		};
+
+		simple-audio-card,dai-link@4 {          /* I2S - AUDIO SYS CODEC 7210_1*/
+				reg = <4>;
+				format = "i2s";
+				cpu {
+						sound-dai = <&i2s_8ch_sd1>;
+				};
+				codec {
+						mclk-fs = <512>;
+						sound-dai = <&es7210_audio_codec_adc1>;
+				};
+		};
+	};
+};
+
+&audio_i2c0 {
+		es8156_audio_codec: es8156@8 {
+			#sound-dai-cells = <0>;
+			compatible = "everest,es8156";
+			reg = <0x08>;
+			sound-name-prefix = "ES8156";
+			AVDD-supply = <&reg_aud_3v3>;
+			DVDD-supply = <&soc_dvdd18_aon_reg>;
+			PVDD-supply = <&soc_dvdd18_aon_reg>;
+			mclk-sclk-ratio = <4>;
+			status = "disabled";
+		};
+
+		es7210_audio_codec_adc0: es7210@40 {
+			#sound-dai-cells = <0>;
+			compatible = "MicArray_0";
+			reg = <0x40>;
+			work-mode = "ES7210_NORMAL_I2S";
+			channels-max = <2>;
+			mclk-sclk-ratio = <4>;
+			sound-name-prefix = "ES7210_ADC0";
+			MVDD-supply = <&reg_aud_3v3>;
+			AVDD-supply = <&reg_aud_3v3>;
+			DVDD-supply = <&soc_dvdd18_aon_reg>;
+			PVDD-supply = <&soc_dvdd18_aon_reg>;
+			status = "disabled";
+		};
+
+		es7210_audio_codec_adc1: es7210@41 {
+			#sound-dai-cells = <0>;
+			compatible = "MicArray_1";
+			reg = <0x41>;
+			work-mode = "ES7210_NORMAL_I2S";
+			channels-max = <2>;
+			mclk-sclk-ratio = <4>;
+			sound-name-prefix = "ES7210_ADC1";
+			MVDD-supply = <&reg_aud_3v3>;
+			AVDD-supply = <&reg_aud_3v3>;
+			DVDD-supply = <&soc_dvdd18_aon_reg>;
+			PVDD-supply = <&soc_dvdd18_aon_reg>;
+			status = "disabled";
+		};
+
+		es7210_adc2: es7210@42 {
+			#sound-dai-cells = <0>;
+			compatible = "MicArray_0";
+			reg = <0x42>;
+			work-mode = "ES7210_TDM_1LRCK_DSPB";
+			channels-max = <8>;
+			sound-name-prefix = "ES7210_ADC2";
+			MVDD-supply = <&reg_aud_3v3>;
+			AVDD-supply = <&reg_aud_3v3>;
+			DVDD-supply = <&soc_dvdd18_aon_reg>;
+			PVDD-supply = <&soc_dvdd18_aon_reg>;
+			status = "disabled";
+		};
+
+		es7210_adc3: es7210@43 {
+			#sound-dai-cells = <0>;
+			compatible = "MicArray_1";
+			reg = <0x43>;
+			work-mode = "ES7210_TDM_1LRCK_DSPB";
+			channels-max = <8>;
+			sound-name-prefix = "ES7210_ADC3";
+			MVDD-supply = <&reg_aud_3v3>;
+			AVDD-supply = <&reg_aud_3v3>;
+			DVDD-supply = <&soc_dvdd18_aon_reg>;
+			PVDD-supply = <&soc_dvdd18_aon_reg>;
+			status = "disabled";
+		};
+
+		audio_aw87519_pa: amp@58 {
+			compatible = "awinic,aw87519_pa";
+			reg = <0x58>;
+			reset-gpio = <&pcal6408ahk_b 3 0x1>;
+			sound-name-prefix = "AW87519";
+			status = "disabled";
+		};
+};
diff --git a/arch/riscv/boot/dts/thead/th1520-a-val.dts b/arch/riscv/boot/dts/thead/th1520-a-val.dts
index 2e5800b12e8e..da1871e8a3c1 100644
--- a/arch/riscv/boot/dts/thead/th1520-a-val.dts
+++ b/arch/riscv/boot/dts/thead/th1520-a-val.dts
@@ -4,3 +4,42 @@
  */
 
 #include "th1520-a-val.dtsi"
+
+&th1520_sound {
+    status = "okay";
+
+    simple-audio-card,dai-link@2 {          /* I2S - HDMI */
+        reg = <2>;
+        format = "i2s";
+        cpu {
+            sound-dai = <&ap_i2s 1>;
+        };
+        codec {
+            sound-dai = <&hdmi_codec>;
+        };
+    };
+};
+
+&ap_i2s {
+	status = "okay";
+};
+
+&i2s0 {
+	status = "okay";
+};
+
+&i2s_8ch_sd2 {
+	status = "okay";
+};
+
+&es8156_audio_codec {
+	status = "okay";
+};
+
+&es7210_audio_codec_adc0 {
+	status = "okay";
+};
+
+&audio_aw87519_pa {
+	status = "okay";
+};
diff --git a/arch/riscv/boot/dts/thead/th1520-a-val.dtsi b/arch/riscv/boot/dts/thead/th1520-a-val.dtsi
index 62c68cb064fe..a8d5c078ad97 100644
--- a/arch/riscv/boot/dts/thead/th1520-a-val.dtsi
+++ b/arch/riscv/boot/dts/thead/th1520-a-val.dtsi
@@ -7,6 +7,7 @@
 
 #include "th1520.dtsi"
 #include <dt-bindings/gpio/gpio.h>
+#include "th1520-a-val-audio.dtsi"
 
 / {
 	model = "XUANTIE TH1520 VAL Board";
@@ -375,6 +376,27 @@ pcal6408ahk_a: gpio@20 {
 	};
 };
 
+&audio_i2c0 {
+	clock-frequency = <100000>;
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&audio_i2c0_pa_pins>;
+};
+
+&audio_i2c1 {
+	clock-frequency = <100000>;
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&audio_i2c1_pa_pins>;
+
+	pcal6408ahk_b: gpio@20 {
+		compatible = "nxp,pcal9554b";
+		reg = <0x20>;
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+};
+
 &mdio0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&mdio0_pins>;
@@ -711,9 +733,72 @@ bt-pins {
 };
 
 &padctrl_aosys {
-	i2s1_pa_pins: i2s1-pa-0 {
+	audio_i2c0_pa_pins: audio-i2c0-pa-0 {
+		i2c-pa-pins {
+			pins = "AUDIO_PA6", "AUDIO_PA7";
+			function = "audio";
+			bias-disable;
+			drive-strength = <0>;
+			slew-rate = <0>;
+		};
+	};
+	audio_i2c1_pa_pins: audio-i2c1-pa-0 {
+		i2c-pa-pins {
+			pins = "AUDIO_PA13", "AUDIO_PA16";
+			function = "audio";
+			bias-disable;
+			drive-strength = <0>;
+			slew-rate = <0>;
+		};
+	};
+	i2s0_pa_pins: i2s0-pa-0 {
+		i2s-pa-pins {
+			pins = "AUDIO_PA9", "AUDIO_PA10", "AUDIO_PA11", "AUDIO_PA12";
+			function = "audio";
+			bias-disable;
+			drive-strength = <0>;
+			slew-rate = <0>;
+		};
+	};
+	i2s_8ch_bus_pa_pins: i2s-8ch-bus-pa-0 {
 		i2s-pa-pins {
-			pins = "AUDIO_PA14", "AUDIO_PA15", "AUDIO_PA16", "AUDIO_PA17";
+			pins = "AUDIO_PA2", "AUDIO_PA3", "AUDIO_PA8";
+			function = "audio";
+			bias-disable;
+			drive-strength = <0>;
+			slew-rate = <0>;
+		};
+	};
+	i2s_8ch_sd0_pa_pins: i2s-8ch-sd0-pa-0 {
+		i2s-pa-pins {
+			pins = "AUDIO_PA4";
+			function = "audio";
+			bias-disable;
+			drive-strength = <0>;
+			slew-rate = <0>;
+		};
+	};
+	i2s_8ch_sd1_pa_pins: i2s-8ch-sd1-pa-0 {
+		i2s-pa-pins {
+			pins = "AUDIO_PA5";
+			function = "audio";
+			bias-disable;
+			drive-strength = <0>;
+			slew-rate = <0>;
+		};
+	};
+	i2s_8ch_sd2_pa_pins: i2s-8ch-sd2-pa-0 {
+		i2s-pa-pins {
+			pins = "AUDIO_PA0";
+			function = "audio";
+			bias-disable;
+			drive-strength = <0>;
+			slew-rate = <0>;
+		};
+	};
+	i2s_8ch_sd3_pa_pins: i2s-8ch-sd3-pa-0 {
+		i2s-pa-pins {
+			pins = "AUDIO_PA1";
 			function = "audio";
 			bias-disable;
 			drive-strength = <0>;
@@ -902,40 +987,8 @@ aon_log_mem: memory@33600000 {
 	};
 };
 
-&audio_i2c0 {
-	clock-frequency = <100000>;
+&ap_i2s {
 	status = "okay";
-
-	es8156_audio_codec: es8156@8 {
-		#sound-dai-cells = <0>;
-		compatible = "everest,es8156";
-		reg = <0x08>;
-		sound-name-prefix = "ES8156";
-		AVDD-supply = <&reg_aud_3v3>;
-		DVDD-supply = <&reg_aud_1v8>;
-		PVDD-supply = <&reg_aud_1v8>;
-	};
-
-	es7210_audio_codec: es7210@40 {
-		#sound-dai-cells = <0>;
-		compatible = "MicArray_0";
-		reg = <0x40>;
-		sound-name-prefix = "ES7210";
-		MVDD-supply = <&reg_aud_3v3>;
-		AVDD-supply = <&reg_aud_3v3>;
-		DVDD-supply = <&reg_aud_1v8>;
-		PVDD-supply = <&reg_aud_1v8>;
-	};
-};
-
-&light_i2s {
-	status = "okay";
-};
-
-&i2s1 {
-	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&i2s1_pa_pins>;
 };
 
 &cpus {
diff --git a/sound/soc/xuantie/th1520-i2s-8ch.c b/sound/soc/xuantie/th1520-i2s-8ch.c
index 97581eff0651..51e194fa306d 100644
--- a/sound/soc/xuantie/th1520-i2s-8ch.c
+++ b/sound/soc/xuantie/th1520-i2s-8ch.c
@@ -434,6 +434,7 @@ static const struct snd_soc_dai_ops th1520_hdmi_dai_ops = {
 
 static struct snd_soc_dai_driver th1520_i2s_8ch_soc_dai[] = {
 	{
+		.name			= "th1520-i2s-8ch-dai",
 		.playback	= {
 			.rates		= TH1520_RATES,
 			.formats	= TH1520_FMTS,
-- 
2.43.0

